web/html/src/manager/systems/virtual-list.tsx

```diff
@@ -0,0 +1,143 @@
+import * as React from "react";
+import { useState, useEffect } from "react";
+import Network from "utils/network";
+import * as Systems from "components/systems";
+import { Utils } from "utils/functions";
+import { Table } from "components/table/Table";
+import { Column } from "components/table/Column";
+import SpaRenderer from "core/spa/spa-renderer";
+import { IconTag } from "components/icontag";
+import { SearchField } from "components/table/SearchField";
+
+// See java/code/src/com/suse/manager/webui/templates/systems/virtual-list.jade
+type Props = {
+  /** Locale of the help links */
+  docsLocale: string;
+  isAdmin: boolean;
+};
+
+function VirtualSystems(props: Props) {
+  const [selectedSystems, setSelectedSystems] = useState([]);
+  const [selectedSystemsCount, setCount] = useState(0);
+
+  const handleSelectedSystems = (data) => {
+    setSelectedSystems(data);
+  };
+
+  useEffect(()=> {
+    setCount(selectedSystems.length);
+    document.getElementById("header_selcount").innerHTML =
+      "<span id='spacewalk-set-system_list-counter' class='badge'>" +
+      selectedSystemsCount.toString() +
+      "</span>" +
+      (selectedSystemsCount === 1 ? "system selected" : "systems selected");
+  }, [handleSelectedSystems]);
+
+  const addToSSM = () => {
```

Comment:
It would be nice to make this method `async` and add error handling, otherwise this request may silently fail and the user won't know. Something along the lines of

```ts
const addToSSM = async () => {
  try {
    await Network.post(...);
  } catch (error) {
    // TODO: Handle error
  }
}
```

Reply:
I think I'll look into that one after addressing all the other comments: this actually is about not using DWR to add systems to SSM.